code/modules/antagonists/obsessed/obsessed.dm

```diff
@@ -0,0 +1,334 @@
+#define OBJECTIVE_STALK "spendtime"
+#define OBJECTIVE_PHOTOGRAPH "polaroid"
+#define OBJECTIVE_STEAL "heirloom"
+#define OBJECTIVE_JEALOUS "jealous"
+
+/datum/antagonist/obsessed
+	name = "Obsessed"
+	show_in_antagpanel = TRUE
+	antagpanel_category = "Other"
+	job_rank = ROLE_OBSESSED
+	show_name_in_check_antagonists = TRUE
+	roundend_category = "obsessed"
+	var/datum/mind/target
+	var/mob/living/carbon/human/human_target
+	var/total_time_stalking
+	var/time_spent_away
+	var/seen_alive = TRUE
+
+/datum/antagonist/obsessed/greet()
+	owner.current.playsound_local(get_turf(owner.current), 'sound/ambience/antag/creepalert.ogg', 100, FALSE, pressure_affected = FALSE, use_reverb = FALSE)
+	to_chat(owner, "<span class='userdanger'>You are the Obsessed!</span>")
+	to_chat(owner, "<B>The Voices have reached out to you, and are using you to complete their evil deeds.</B>")
+	to_chat(owner, "<B>You don't know their connection, but The Voices compel you to stalk [target], forcing them into a state of constant paranoia.</B>")
+	to_chat(owner, "<B>The Voices will retaliate if you fail to complete your tasks or spend too long away from your target.</B>")
+	to_chat(owner, "<span class='boldannounce'>This role does NOT enable you to otherwise surpass what's deemed creepy behavior per the rules.</span>")//ironic if you know the history of the antag
+	owner.current.client?.tgui_panel?.give_antagonist_popup("Obsession",
+		"Stalk [human_target.real_name] and force them into a constant state of paranoia.")
+
+/datum/antagonist/obsessed/on_gain()
+	find_target()
+	if(!target)
+		qdel(src)
+		return
+	//We need to find target first before calling parent here
+	. = ..()
+	forge_objectives()
+	RegisterSignal(owner.current, COMSIG_GLOB_MOB_DEATH, .proc/OnDeath)
```

Comment:
I think this is the wrong signal. Maybe instead handle this by returning `PROCESS_KILL` in `process` if the mob is dead?

Reply:
I don't want to check every tick if the mob's alive or not. Returning process kill is essentially the same as STOP_PROCESSING

Reply:
Regardless, this is a global signal, as far as I know the code won't work as it is.

Reply:
?
Why wouldn't it work?